pkg/app/api/grpcapi/piped_api.go

```diff
@@ -886,6 +886,20 @@ func (a *PipedAPI) PutLatestAnalysisResult(ctx context.Context, req *pipedservic
 	return &pipedservice.PutLatestAnalysisResultResponse{}, nil
 }
 
+func (a *PipedAPI) GetDesiredVersion(ctx context.Context, _ *pipedservice.GetDesiredVersionRequest) (*pipedservice.GetDesiredVersionResponse, error) {
+	_, pipedID, _, err := rpcauth.ExtractPipedToken(ctx)
+	if err != nil {
+		return nil, err
+	}
+	piped, err := getPiped(ctx, a.pipedStore, pipedID, a.logger)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
yeah, it's fine to check if DesiredVersion is empty or not at the caller (Piped) side.